From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: wosnxn123 <wosnxn123@users.noreply.github.com>
Date: Thu, 7 Aug 2026 12:00:00 +0800
Subject: [PATCH] Spawn invulnerable time

Ported from Lophine 0096 (MIT, Helvetica Volubi).

diff --git a/net/minecraft/server/level/ServerPlayer.java b/net/minecraft/server/level/ServerPlayer.java
index c19c621b603aab2e85ba9b0fa98e7b270fdf1358..899e224b29d7758df31c6e3a11634b2431d35a98 100644
--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -243,6 +243,7 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
     private int lastSentFood = -99999999;
     private boolean lastFoodSaturationZero = true;
     public int lastSentExp = -99999999;
+    protected int spawnInvulnerableTime = 60; // Lophine - spawn invulnerable time
     private ChatVisiblity chatVisibility = ChatVisiblity.FULL;
     public ParticleStatus particleStatus = ParticleStatus.ALL;
     private boolean canChatColor = true;
@@ -958,6 +959,7 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
         this.connection.tickClientLoadTimeout();
         this.gameMode.tick();
         this.wardenSpawnTracker.tick();
+        if (this.spawnInvulnerableTime > 0) --this.spawnInvulnerableTime; // Lophine - spawn invulnerable time
         if (this.invulnerableTime > 0) {
             this.invulnerableTime--;
         }
@@ -1538,6 +1540,13 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
             return false;
         }
 
+        // Lophine start - spawn invulnerable time
+        if (io.canvasmc.canvas.GlobalConfiguration.getInstance().oldFeature.spawnInvulnerableTime) {
+            if (this.spawnInvulnerableTime > 0 && !source.is(net.minecraft.tags.DamageTypeTags.BYPASSES_INVULNERABILITY)) {
+                return false;
+            }
+        }
+        // Lophine end - spawn invulnerable time
         Entity entity = source.getEntity();
         if (!( // Paper - split the if statement. If below statement is false, hurtServer would not have been evaluated. Return false.
             !(entity instanceof Player player && !this.canHarmPlayer(player))
